Classes/Domain/Repository/HaikuRepository.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\Domain\Repository;
+
+use TYPO3\CMS\Core\Database\ConnectionPool;
+
+class HaikuRepository
+{
+    public function __construct(
+        protected readonly ConnectionPool $connectionPool,
```

Comment:
You can inject the QueryBuilder directly.

First, use the factory in Services.yaml like here:
https://github.com/brotkrueml/typo3-jobrouter-data/blob/main/Configuration/Services.yaml#L11
The argument is the table name.

Second, inject it into the repository like here:
https://github.com/brotkrueml/typo3-jobrouter-data/blob/main/Configuration/Services.yaml#L96
As argument use "$queryBuilder".

Then you can use "protected readonly QueryBuilder $queryBuilder"

Let DI make the work.

Reply:
The QueryBuilder may only be used once per Query. Therefore it has to be created each time you are doing a query: https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Database/QueryBuilder/Index.html#instantiation "The QueryBuilder holds internal state and should not be re-used for different queries: Use one query builder per query."

Reply:
Mea culpa, but you can inject the Connection that way instead the ConnectionPool.